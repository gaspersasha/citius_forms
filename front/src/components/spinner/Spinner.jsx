import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import s from './spinner.module.sass';

const Spinner = ({ className, ...rest }) => (
  <div className={s.container} {...rest}>
    <div className={cn(s.loader, className)} />
  </div>
);

Spinner.propTypes = {
  className: PropTypes.string,
};
Spinner.defaultProps = {
  className: '',
};

export default Spinner;
