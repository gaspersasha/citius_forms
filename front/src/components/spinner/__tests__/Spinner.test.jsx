import React from 'react';
import Spinner from '../Spinner';

describe('<Spinner />', () => {
  let wrapper;

  const getWrapper = (props) => shallow(<Spinner {...props} />);

  it('renders without errors', () => {
    wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });
});
