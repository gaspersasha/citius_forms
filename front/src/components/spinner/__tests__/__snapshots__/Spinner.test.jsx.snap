// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders without errors 1`] = `
<div
  className="container"
>
  <div
    className="loader"
  />
</div>
`;
